From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: HaHaWTH <102713261+HaHaWTH@users.noreply.github.com>
Date: Tue, 9 Nov 2077 00:00:00 +0800
Subject: [PATCH] GUI theme

License: GPL-3.0-only (https://www.gnu.org/licenses/gpl-3.0.html)

Additional term under GNU GPLv3 section 7(b):

Any copy or modified version containing a substantial portion of the
original contributions in this patch must preserve the following
author attribution in the patch header, source code, or another
prominent legal notice accompanying the corresponding source:

Original patch by HaHaWTH <102713261+HaHaWTH@users.noreply.github.com>

diff --git a/net/minecraft/server/gui/MinecraftServerGui.java b/net/minecraft/server/gui/MinecraftServerGui.java
index db4d0cc0149b54e5b8a9b3e646d690e6da7cf782..537af6eec7e7f2460012ad082b4d09965f661635 100644
--- a/net/minecraft/server/gui/MinecraftServerGui.java
+++ b/net/minecraft/server/gui/MinecraftServerGui.java
@@ -49,7 +49,7 @@ public class MinecraftServerGui extends JComponent {
 
     public static MinecraftServerGui showFrameFor(final DedicatedServer server) {
         try {
-            UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
+            org.dreeam.leaf.gui.GuiTheme.install(); // Leaf - GUI theme
         } catch (Exception var3) {
         }
 
@@ -58,6 +58,7 @@ public class MinecraftServerGui extends JComponent {
         frame.setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE);
         frame.add(gui);
         frame.pack();
+        org.dreeam.leaf.gui.GuiTheme.applyWindowTheme(frame); // Leaf - GUI theme
         frame.setLocationRelativeTo(null);
         frame.setVisible(true);
         // Paper start - Improve ServerGUI
@@ -113,6 +114,7 @@ public class MinecraftServerGui extends JComponent {
     private JComponent buildPlayerPanel() {
         JList<?> playerList = new PlayerListComponent(this.server);
         JScrollPane scrollPane = new JScrollPane(playerList, ScrollPaneConstants.VERTICAL_SCROLLBAR_ALWAYS, ScrollPaneConstants.HORIZONTAL_SCROLLBAR_AS_NEEDED);
+        org.dreeam.leaf.gui.GuiTheme.styleScrollPane(scrollPane); // Leaf - GUI theme
         scrollPane.setBorder(new TitledBorder(new EtchedBorder(), "Players"));
         return scrollPane;
     }
@@ -121,6 +123,7 @@ public class MinecraftServerGui extends JComponent {
         JPanel panel = new JPanel(new BorderLayout());
         org.purpurmc.purpur.gui.JColorTextPane chatArea = new org.purpurmc.purpur.gui.JColorTextPane(); // Purpur - GUI Improvements
         JScrollPane scrollPane = new JScrollPane(chatArea, ScrollPaneConstants.VERTICAL_SCROLLBAR_ALWAYS, ScrollPaneConstants.HORIZONTAL_SCROLLBAR_AS_NEEDED);
+        org.dreeam.leaf.gui.GuiTheme.styleScrollPane(scrollPane); // Leaf - GUI theme
         chatArea.setEditable(false);
         chatArea.setFont(MONOSPACED);
         JTextField chatField = new JTextField();
diff --git a/org/purpurmc/purpur/gui/GUIColor.java b/org/purpurmc/purpur/gui/GUIColor.java
index 550222758bf0e7deff26a6e813a860b7be365e87..54ad72f7130feaf6fd7377bd19ff680057363311 100644
--- a/org/purpurmc/purpur/gui/GUIColor.java
+++ b/org/purpurmc/purpur/gui/GUIColor.java
@@ -7,35 +7,39 @@ import java.util.HashMap;
 import java.util.Map;
 
 public enum GUIColor {
-    BLACK(ChatColor.BLACK, new Color(0x000000)),
-    DARK_BLUE(ChatColor.DARK_BLUE, new Color(0x0000AA)),
-    DARK_GREEN(ChatColor.DARK_GREEN, new Color(0x00AA00)),
-    DARK_AQUA(ChatColor.DARK_AQUA, new Color(0x009999)),
-    DARK_RED(ChatColor.DARK_RED, new Color(0xAA0000)),
-    DARK_PURPLE(ChatColor.DARK_PURPLE, new Color(0xAA00AA)),
-    GOLD(ChatColor.GOLD, new Color(0xBB8800)),
-    GRAY(ChatColor.GRAY, new Color(0x888888)),
-    DARK_GRAY(ChatColor.DARK_GRAY, new Color(0x444444)),
-    BLUE(ChatColor.BLUE, new Color(0x5555FF)),
-    GREEN(ChatColor.GREEN, new Color(0x55FF55)),
-    AQUA(ChatColor.AQUA, new Color(0x55DDDD)),
-    RED(ChatColor.RED, new Color(0xFF5555)),
-    LIGHT_PURPLE(ChatColor.LIGHT_PURPLE, new Color(0xFF55FF)),
-    YELLOW(ChatColor.YELLOW, new Color(0xFFBB00)),
-    WHITE(ChatColor.WHITE, new Color(0xBBBBBB));
+    // Leaf start - GUI theme
+    BLACK(ChatColor.BLACK, new Color(0x000000), new Color(0x9299A3)),
+    DARK_BLUE(ChatColor.DARK_BLUE, new Color(0x0000AA), new Color(0x7AA2F7)),
+    DARK_GREEN(ChatColor.DARK_GREEN, new Color(0x00AA00), new Color(0x78BE8C)),
+    DARK_AQUA(ChatColor.DARK_AQUA, new Color(0x009999), new Color(0x63C5C8)),
+    DARK_RED(ChatColor.DARK_RED, new Color(0xAA0000), new Color(0xEF7078)),
+    DARK_PURPLE(ChatColor.DARK_PURPLE, new Color(0xAA00AA), new Color(0xC792EA)),
+    GOLD(ChatColor.GOLD, new Color(0xBB8800), new Color(0xE9BD69)),
+    GRAY(ChatColor.GRAY, new Color(0x888888), new Color(0xB6BECA)),
+    DARK_GRAY(ChatColor.DARK_GRAY, new Color(0x444444), new Color(0x9299A3)),
+    BLUE(ChatColor.BLUE, new Color(0x5555FF), new Color(0x8CAFFF)),
+    GREEN(ChatColor.GREEN, new Color(0x55FF55), new Color(0x98D98E)),
+    AQUA(ChatColor.AQUA, new Color(0x55DDDD), new Color(0x89DDDF)),
+    RED(ChatColor.RED, new Color(0xFF5555), new Color(0xFF8990)),
+    LIGHT_PURPLE(ChatColor.LIGHT_PURPLE, new Color(0xFF55FF), new Color(0xDBA4EF)),
+    YELLOW(ChatColor.YELLOW, new Color(0xFFBB00), new Color(0xF0D078)),
+    WHITE(ChatColor.WHITE, new Color(0xBBBBBB), new Color(0xE4E7EC));
+    // Leaf end - GUI theme
 
     private final ChatColor chat;
     private final Color color;
+    private final Color darkColor; // Leaf - GUI theme
 
     private static final Map<ChatColor, GUIColor> BY_CHAT = new HashMap<>();
 
-    GUIColor(ChatColor chat, Color color) {
+    GUIColor(ChatColor chat, Color color, Color darkColor) { // Leaf - GUI theme
         this.chat = chat;
         this.color = color;
+        this.darkColor = darkColor; // Leaf - GUI theme
     }
 
     public Color getColor() {
-        return color;
+        return org.dreeam.leaf.gui.GuiTheme.isDark() ? darkColor : color; // Leaf - GUI theme
     }
 
     public ChatColor getChatColor() {
diff --git a/org/purpurmc/purpur/gui/JColorTextPane.java b/org/purpurmc/purpur/gui/JColorTextPane.java
index d75fb5e77eff27d86135ed7d605dbc250b660f7d..26e1b78d50c10987b03ace8e275a38c171982e9a 100644
--- a/org/purpurmc/purpur/gui/JColorTextPane.java
+++ b/org/purpurmc/purpur/gui/JColorTextPane.java
@@ -17,7 +17,7 @@ import java.util.Set;
 public class JColorTextPane extends JTextPane {
     private static final GUIColor DEFAULT_COLOR;
     static {
-        DEFAULT_COLOR = UIManager.getSystemLookAndFeelClassName().equals("com.sun.java.swing.plaf.gtk.GTKLookAndFeel")
+        DEFAULT_COLOR = org.dreeam.leaf.gui.GuiTheme.isDark() // Leaf - GUI theme
             ? GUIColor.WHITE : GUIColor.BLACK;
     }
 
